doc/migration.md

```diff
@@ -0,0 +1,93 @@
+# Hypersync SDK Migration Guide
```

Comment:
nit: filename could be referencing 2->3 migration so when we go 3->4 we have consistent naming

Reply:
The idea is that there will only be one migration guide an that each version to version step will have a section.  This is consistent with other migration guides I have seen.